Xception/xception.py

```diff
@@ -177,3 +177,188 @@ def forward(self,x):
+        self.sepconv2 = DepthwiseSeparable(in_channel=out_channel,out_channel=out_channel,kernel_size=3)
+        self.sepconv3 = DepthwiseSeparable(in_channel=out_channel,out_channel=out_channel,kernel_size=3)
+        self.relu = nn.ReLU()
+
+        # 2nd branch
+        self.conv = nn.Conv2d(in_channels=in_channel,out_channels=out_channel,kernel_size=1,stride=1)
+
+    def forward(self,x):
+        # 2nd branch
+        y = self.conv(x)
+
+        # 1st branch
+        x = self.relu(x)
+        x = self.sepconv1(x)
+
+        x = self.relu(x)
+        x = self.sepconv2(x)
+
+        x = self.relu(x)
+        x = self.sepconv3(x)
+
+        # Add two branch
+        x = x + y
+        return x
+
+
+# In[12]:
+
+
+model = MiddleflowSeperable(in_channel=728,out_channel=728)
+summary(model=model,input_size=(728,19,19))
+
+
+# In[13]:
+
+
+class MiddleFlow(nn.Module):
+    """
+    This is the Middle Flow part -
+        MiddleFlowSeperable is repeated 8 times 
+    
+    input_size = (19,19,728)
+    output_size = (19,19,728)       
+    """
+    def __init__(self):
+        super(MiddleFlow,self).__init__()
+        self.sep = MiddleflowSeperable(in_channel=728,out_channel=728)
+    def forward(self,x):
+        for i in range(0,8):
+            x = self.sep(x)
+        return x
+
+
+# In[14]:
+
+
+sample = MiddleFlow()
+summary(model=sample,input_size=(728,19,19))
+
+
+# In[15]:
+
+
+class ExitflowSeperable(nn.Module):
+    """
+    This part contains depthwise separable convolutions and is repeated 2 times in original implementation with max pool layer.
+
+        in_channel, out_channel : Both of them are different
+        kernel_size = 3 : For all repetitions
+        max pool kernel_size :3 with stride:2
+    """
+    def __init__(self,in_channel,out_channel,padding=1):
+        super(ExitflowSeperable,self).__init__()
+
+        #1st branch
+        self.sep1 = DepthwiseSeparable(in_channel=in_channel,out_channel=in_channel,kernel_size=3)
+        self.sep2 = DepthwiseSeparable(in_channel=in_channel,out_channel=out_channel,kernel_size=3)
+        self.relu = nn.ReLU()
+        self.pool = nn.MaxPool2d(kernel_size=3,stride=2,padding=padding)
+        
+        #2nd branch
+        self.conv = nn.Conv2d(in_channels=in_channel,out_channels=out_channel,kernel_size=1,stride=2)
+
+    def forward(self,x):
+
+        #2nd branch 
+        y = self.conv(x)
+
+        #1st branch
+        x = self.relu(x)
+        x = self.sep1(x)
+        x = self.relu(x)
+        x = self.sep2(x)
+        x = self.pool(x)
+
+        return x+y    
+
+
+# In[16]:
+
+
+sample = ExitflowSeperable(in_channel=728,out_channel=1024)
+summary(model=sample,input_size=(728,19,19))
+
+
+# In[17]:
+
+
+class ExitFlow(nn.Module):
+    """
+    This part contains ExitFlowSeperable part with 2 different depthwise seperable convolutions followed by Global Avgerage Pool(Avg Pool of kernel size 10) and connecting with output layer
+
+    input_size  :(19,19,728)
```

Comment:
(728,19,19) or 19 x 19 x 728

Reply:
(728,19,19) is equivalent to 19x19x728 in pytorch

Reply:
It will be better Using (728,19,19) in place of (19,19,728) If size is written in tuple form,